### ISSUE TITLE
[<Ray component: Train>] OOM when loading 70B LLAMA2 

### ISSUE THREAD

**Issue by robinsonmhj** [ray-project/ray#43318]
### What happened + What you expected to happen

get OOM when loading llama2 70b from the line below

model = AutoModelForCausalLM.from_pretrained(
        pretrained_path,
        trust_remote_code=True,
        torch_dtype=torch.bfloat16,
        # `use_cache=True` is incompatible with gradient checkpointing.
        use_cache=False,
        use_flash_attention_2=True,
    )

### Versions / Dependencies

ray[all]==2.9.2

### Reproduction script

https://github.com/ray-project/ray/blob/master/doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py#L284

it always get OOM during this line of code although I have 670GB CPU memory

model = AutoModelForCausalLM.from_pretrained(
        pretrained_path,
        trust_remote_code=True,
        torch_dtype=torch.bfloat16,
        # `use_cache=True` is incompatible with gradient checkpointing.
        use_cache=False,
        use_flash_attention_2=True,
    )

### Issue Severity

High: It blocks me from completing my task.

**Comment by robinsonmhj** [comment]
I am using the above example on 5 node with 8*A100(32GB GPU Memory) and 92 CPU(RAM 670GB), I always get OOM on CPU. Based on the ray document, [training_fucntion](https://github.com/ray-project/ray/blob/master/doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py#L219) needs to run on each worker, and each worker has 1 GPU and they share the CPU and RAM on the node, which means that there are 8 workers per node see the [configuration](https://github.com/ray-project/ray/blob/master/doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py#L759). For the llama2 70b model, it needs 720GB RAM to load the model, see the calculation below, 
```
Model parameters: 70(billion parameters) * 2(FP16) ≈ 160GB
Optimizer states: 70(billion parameters) * 2(momentums per param) * 4 (FP32) ≈ 560GB
```
The calculation above is based on this [ray document](https://docs.ray.io/en/latest/train/examples/lightning/vicuna_13b_lightning_deepspeed_finetune.html)

Here are my questions:

1. Does each worker run the [training_fucntion](https://github.com/ray-project/ray/blob/master/doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py#L219)? 
1.1) if yes, how can the model loaded into the CPU using g5.48xlarge instance as there is only 760GB RAM? Because there are 8 workers on each node, it needs 560GB*8=4480GB, because AutoModelForCausalLM.from_pretrained() load the model into CPU. 
1.2) if no. Which worker run the function? How does the data shared between workers, in this case, the data is the model?

**Comment by robinsonmhj** [comment]
@kouroshHakha and @ArturNiederfahrenhorst Can help take a look or answer my questions?
